astropy/timeseries/downsample.py

```diff
@@ -0,0 +1,136 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+import warnings
+
+import numpy as np
+from astropy import units as u
+from astropy.utils.exceptions import AstropyUserWarning
+
+from astropy.timeseries.sampled import TimeSeries
+from astropy.timeseries.binned import BinnedTimeSeries
+
+__all__ = ['simple_downsample']
+
+
+def reduceat(array, indices, function):
+    """
+    Manual reduceat functionality for cases where Numpy functions don't have a reduceat.
+    It will check if the input function has a reduceat and call that if it does.
+    """
+    if hasattr(function, 'reduceat'):
+        return np.array(function.reduceat(array, indices))
+    else:
+        result = []
+        for i in range(len(indices) - 1):
+            if indices[i+1] <= indices[i]+1:
+                result.append(function(array[indices[i]]))
+            else:
+                result.append(function(array[indices[i]:indices[i+1]]))
+        result.append(function(array[indices[-1]:]))
+        return np.array(result)
+
+
+def simple_downsample(time_series, time_bin_size, func=None, time_bin_start=None, n_bins=None):
```

Comment:
I am not a fan of the name `simple_downsample`, as I think it's not really useful to call anything "simple" (both because "simple" is often surprisingly complex, and "simple" isn't really that descriptive).  

What about `binned_downsample`?  Or `uniform_downsample`? (depending on what you think about my next comment...)

Reply:
What about ``aggregate_downsample`` since it's basically just applying aggregate functions to each bin?

Reply:
Renamed to ``aggregate_downsample`` following Slack discussion

Reply:
I'll save my suggestion to rename to `agitate_downsample` for next Apr. 😉